nise/report.py

```diff
@@ -501,33 +501,39 @@ def _create_generator_dates_from_yaml(attributes, month):
     gen_start_date = None
     gen_end_date = None
 
+    start_date = attributes.get("start_date")
+    end_date = attributes.get("end_date")
+    month_start = month.get("start")
+    month_end = month.get("end")
+
+    # Use a different variable for the end of month comparison. This matters
+    # when the end_date is the same as the month_end, which can happen based
+    # on the specified --end-date parameter or on the first of the month.
+    month_end_compare = month_end
+    if (month_end.month, month_end.day) != (end_date.month, end_date.day):
+        # Create a new datetime object and store it, leaving the original
+        # month_end object unmodified.
+        month_end_compare = month_end_compare.replace(hour=23, minute=59, second=59)
+
     # Generator range is larger then current month on both start and end
-    if attributes.get("start_date") < month.get("start") and attributes.get("end_date") > month.get("end").replace(
-        hour=23, minute=59, second=59
-    ):
-        gen_start_date = month.get("start")
-        gen_end_date = month.get("end")
+    if start_date < month_start and end_date > month_end_compare:
+        gen_start_date = month_start
+        gen_end_date = month_end
 
     # Generator starts before month start and ends within month
-    if attributes.get("start_date") <= month.get("start") and attributes.get("end_date") <= month.get("end").replace(
-        hour=23, minute=59, second=59
-    ):
-        gen_start_date = month.get("start")
-        gen_end_date = attributes.get("end_date")
+    elif start_date <= month_start and end_date <= month_end_compare:
```

Comment:
not a big deal, but I would use `start_date < month_start `  instead of `start_date <= month_start`  because the next elif includes start_date == month_start: `start_date >= month_start`

Reply:
I was thinking these conditionals needed some reworking but I didn't want to introduce more bugs and I had been staring at it too long. I'll go ahead and make this change. Thanks for the suggestion.